### DISCUSSION TITLE
Nginx Proxy Manager Restore Backup from Docker

### DISCUSSION THREAD

**Discussion by dn4hc** [tteck/Proxmox#2451] [Help Needed]
Looks like I selected the wrong category so here it is again in Discussions. 

Aim is to move to lxc from docker and Nginx Proxy Manager Restore config from Docker. 

I’m trying to export or backup from docker to the lxc npm installed using your script however i’m not able to restore it successfully. NPM always acts as fresh installed.  Possible to know the mapped folder to restore the docker config volume {data, lets encrypt}?

Thanks in advance

**Comment by X4V1** [comment]
Hello,

I'm also trying to migrate from docker to proxmox LXC.

I was mapping 2 volumes to my docker:
      - ./data:/data
      - ./letsencrypt:/etc/letsencrypt


so I tried to reproduce it in the lxc container with no luck:
- stop npm and openresty services
- replace /data folder the one from docker
- replace /etc/letsencrypt with the one from docker
- restart npm and openresty service but openresty is not starting anymore

I got the following error:
```
Feb 08 15:07:45 nginxproxymanager systemd[1]: Starting openresty.service - The OpenResty Application Platform...
Feb 08 15:07:45 nginxproxymanager nginx[170]: nginx: [emerg] open() "/usr/local/openresty/nginx/conf/conf.d/include/letsencrypt-acme-challenge.conf" failed (2: No such file or directory) in /data/nginx/default_host/site.conf:14
Feb 08 15:07:45 nginxproxymanager nginx[170]: nginx: configuration file /usr/local/openresty/nginx/conf/nginx.conf test failed
Feb 08 15:07:45 nginxproxymanager systemd[1]: openresty.service: Control process exited, code=exited, status=1/FAILURE
Feb 08 15:07:45 nginxproxymanager systemd[1]: openresty.service: Failed with result 'exit-code'.
Feb 08 15:07:45 nginxproxymanager systemd[1]: Failed to start openresty.service - The OpenResty Application Platform.
```

it complains about a missing file: `/usr/local/openresty/nginx/conf/conf.d/include/letsencrypt-acme-challenge.conf` (referenced from `/data/nginx/default_host/site.conf` that was copied from the docker backup)

Content of `/data/nginx/default_host/site.conf`:
```
# ------------------------------------------------------------
# Default Site
# ------------------------------------------------------------

server {
  listen 80 default;
listen [::]:80 default;

  server_name default-host.localhost;
  access_log /data/logs/default-host_access.log combined;
  error_log /data/logs/default-host_error.log warn;


  include conf.d/include/letsencrypt-acme-challenge.conf;
  location / {
    return 404;
  }

}
```

the `include conf.d/include/letsencrypt-acme-challenge.conf;` seems to use a relative path. 
so I tried to compare where that file is located:
- file location on a in a docker container (not using lxc, proxmox, ... just a simple docker container on a debian os): `/etc/nginx/conf.d/include/letsencrypt-acme-challenge.conf`
- file location in the npm lxc: `/usr/local/openresty/nginx/conf.d/include/letsencrypt-acme-challenge.conf`

it fails to load: `/usr/local/openresty/nginx/conf/conf.d/include/letsencrypt-acme-challenge.conf`
but this one exists: `/usr/local/openresty/nginx/conf.d/include/letsencrypt-acme-challenge.conf`

I think there is something wrong because have conf.d inside the conf directory does not seems correct to me. But I don't know why it takes that path.

I'm still investigating but I don't really know where to continue. I tried to read the script from @tteck to see what could be the difference but I don't see anything changing that path. If someone has an idea, please let me know.

**Reply by bilogic** [reply]
For this error, I symlinked the source `/usr/local/openresty/nginx/conf.d` to `/usr/local/openresty/nginx/conf/conf.d`, but faced other errors

**Reply by bilogic** [reply]
how did you manage to get your logs? mine is always empty, 0 bytes

```
root@nginxproxymanager:/data/logs# ls -l
total 8
-rw-r--r-- 1 root root    0 Jul 15 11:59 fallback_access.log
-rw-r--r-- 1 root root 1303 Jul 15 12:14 fallback_error.log
-rw-r--r-- 1 root root    0 Jul 15 12:06 letsencrypt-requests_access.log
-rw-r--r-- 1 root root    0 Jul 15 12:06 letsencrypt-requests_error.log
-rw-r--r-- 1 root root    0 Jul 15 12:14 proxy-host-1_access.log
-rw-r--r-- 1 root root    0 Jul 15 12:14 proxy-host-1_error.log
-rw-r--r-- 1 root root    0 Jul 15 12:07 proxy-host-28_access.log
-rw-r--r-- 1 root root    0 Jul 15 12:07 proxy-host-28_error.log
```

**Comment by dn4hc** [comment]
@tteck any help will be appreciated.

**Reply by aelsharawi** [reply]
https://dannyda.com/2020/08/26/how-to-passthrough-hdd-ssd-physical-disks-to-vm-on-proxmox-vepve/

I used this to passthrough to proxmox (is this ok?)

will keep proxmox for now (so as to keep using your scripts :) )

**Comment by bilogic** [comment]
Together with importing from Docker I think it also makes lot of sense to be able to backup just the few megabytes of critical data instead of 4GB of LXC container, I wonder why the LXC container did not have mount points built-in.
